instock_connector/scripts/install_service.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+# Installs a systemd service for the instock-connector
+
+
+if [ $# -eq 0 ]; then
+    echo "Usage: $0 [--uninstall] <config_basename>.<robot_id>"
+    echo "It will create and enable /etc/systemd/system/instock-connector@<config_basename>.<robot_id>.service"
+    echo "The service will run the InOrbit Instock Connector the <config_basename>.yaml configuration and select the <robot_id> robot"
+    echo "If --uninstall is passed it will stop and remove the service"
+    exit 1
+fi
+
+echo This script will run commands with sudo. You may be prompted for your password
+read -n 1 -r -p "Do you wish to continue? [Y/n]" ans;
+case "$ans" in
+    n|N)
+        exit 1;;
+    Y|y|"")
+        ;;
+    *)
+        exit 1;;
+esac
+
+NAME=$1
+TEMPLATE="$( realpath $( dirname $0 ) )/instock-connector@.service"
+SYSTEMD_SERVICE_NAME="instock-connector@$1.service"
+START_SCRIPT_LOCATION="$( realpath $( dirname $0 ) )/start.sh"
+DESTINATION_START_SCRIPT="/usr/local/bin/instock-connector.sh"
+
+# Uninstall procedure
+if [ "$1" == "--uninstall" ]; then
+    if [ $# -ne 2 ]; then
+        echo "Usage: $0 --uninstall <service_name>"
+        echo Missing service name
+        exit 1
+    fi
+
+    echo Uninstalling systemd service instock-connector@$2.service
+    sudo systemctl stop instock-connector@$2.service
+    sudo systemctl disable instock-connector@$2.service
+    sudo rm /etc/systemd/system/instock-connector@$2.service
+    sudo systemctl daemon-reload
+
+    echo Removing $DESTINATION_START_SCRIPT
+    sudo rm $DESTINATION_START_SCRIPT
+    exit 0
+fi
+
+# Check if the inorbit user exists
+if ! id inorbit >/dev/null 2>&1; then
+    echo \`inorbit\` user not found. Creating it...
+    sudo useradd inorbit
```

Comment:
Is there a reason for creating a user instead of defaulting to root when the service is installed?

Reply:
I removed the bit that creates a user and instead let the script user know that a certain user is expected and that it should be created or modified in the template before proceeding with the installation.

I looked into using `User=nobody` (considered unsafe) or `DynamicUser=true` (prevents filesystem access to home and changes to are not preserved). More information in the internal issue.

Reply:
Great investigation!